js/page.js

```diff
@@ -290,17 +298,40 @@
       for (let site of gGrid.sites) {
         if (site && site._link.url === message.link.url) {
           site._link.pinState = message.pinState;
+          break;
         }
       }
+      // If we are unpinning the site, update the grid.
+      if (!message.pinState) {
+        gUpdater.updateGrid(message);
+      }
     },
 
     setBlockState(message) {
       for (let site of gGrid.sites) {
         if (site && site._link.url === message.link.url) {
           site._link.blockState = message.blockState;
+          break;
         }
       }
+      // If we are blocking the site, update the grid.
+      if (message.blockState) {
+        gUpdater.updateGrid(message);
+      }
     },
+
+    /**
+     * Find the correct URL to display.
+     *
+     * @param {Object} aURL The current URL sent down from the parent.
+     */
+    findURL(aURL) {
```

Comment:
The name of this method is a bit confusing. I would expect a `find` method to return me something. It is loading? the URL?  What does "find" mean in this context?